### DISCUSSION TITLE
Welcome to Propan Discussions!

### DISCUSSION THREAD

**Discussion by Lancetnik** [Lancetnik/Propan#4] [General]
# :wave: Welcome!

We’re using Discussions as a place to connect with other members of our community. We hope that you:

* Ask questions you’re wondering about.
* Share ideas.
* Engage with other community members.
* Welcome others and are open-minded. Remember that this is a community we
build together :muscle:.

To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bodograumann** [comment]
Hi, I just wanted to say that I built something similar earlier this year.
I wish I had known about this project then. It sounds really exciting.

**Reply by Lancetnik** [reply]
Thanks for your feedback! With honesty, this project become the current state since this april. Earlier it wasn't so `exciting`.

**Comment by abxsantos** [comment]
Hello!
I've recently started using your project (great job by the way!) and was wondering if there's a way to run a specific handler through a Propan command.

Ex.:
My application currently has these two handlers:
```python
@router.event("test")
async def base_handler(body: str):
    logger.info(f"Received message: {body}")

@router.event("test-2")
async def base_handler_two(body: str):
    logger.info(f"Received message: {body}")
```
But I want to run them separately (so that I can better manage scalability). Is there any way this can be achieved?

**Reply by Lancetnik** [reply]
Hi! Thanks for your kind feedback. The most relevant way I thinking about: separate your handlers between different routers and include them in `app.on_startup` hook based on your custom command-line options.

Smth, looking loke this:

```python
from typing import Literal
from propan import PropanApp, RabbitBroker, RabbitRouter

broker = RabbitBroker()
app = PropanApp(broker)

router1 = RabbitRouter()

@router1.handle("test-1")
async def handler():
    pass

router2 = RabbitRouter()

@router2.handle("test2")
async def handler2():
    pass

@app.on_startup
def setup_routers(version: Literal["both", "1", "2"] = "both"):
    if version == "both" or version == "1":
        broker.include_router(router1)
    if version == "both" or version == "2":
        broker.include_router(router2)
```

If you are using the **Propan** as a **FastAPI** router, you able to make smth closer (or just separate it into 2 different scripts), but based on your project settings.

**Reply by abxsantos** [reply]
Great to know!!! Thanks a lot for the quick response!
